railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -432,7 +432,7 @@ def rails_app_path
       end
 
       def inside_application?
-        rails_app_path && app_path =~ /^#{rails_app_path}/
+        rails_app_path && destination_root =~ /^#{rails_app_path}/
```

Comment:
Think we can spare the `&&` with: `destination_root.start_with?(rails_app_path.to_s)`